src/src/com/microsoft/aad/adal/Discovery.java

```diff
@@ -137,10 +137,10 @@ private void addValidHostToList(URL validhost) {
     private void initValidList() {
         // mValidHosts is a sync set
         if (sValidHosts.size() == 0) {
-            sValidHosts.add("login.windows.net");
-            sValidHosts.add("login.microsoftonline.com");
-            sValidHosts.add("login.chinacloudapi.cn");
-            sValidHosts.add("login.cloudgovapi.us");
+            sValidHosts.add("login.windows.net"); // Microsoft Azure Worldwide - Used in validation scenarios where host is not this list 
+            sValidHosts.add("login.microsoftonline.com"); // Microsoft Azure Worldwide
+            sValidHosts.add("login.chinacloudapi.cn"); // Microsoft Azure China
+            sValidHosts.add("login-us.microsoftonline.com"); // Microsoft Azure US Government
```

Comment:
Why don't you add the German one here as well. Just trying to prevent 2 one line code reviews.